Signal/src/ViewControllers/AddToBlockListViewController.m

```diff
@@ -332,6 +375,113 @@ - (BOOL)textFieldShouldReturn:(UITextField *)textField
     return NO;
 }
 
+#pragma mark - UITableViewDataSource
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+    return (NSInteger)self.contacts.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    Contact *contact = self.contacts[(NSUInteger)indexPath.item];
+
+    ContactTableViewCell *cell = [_contactsTableView cellForRowAtIndexPath:indexPath];
+    if (!cell) {
+        cell = [ContactTableViewCell new];
+    }
+    [cell configureWithContact:contact contactsManager:self.contactsManager];
```

Comment:
Forgive the stream of consciousness:

It feels a little strange when you block a user and they're still in this list (and then you can "block them again"). 

I think this causes it to be unclear that blocking "worked" (Yes, **even though** there is a "success" alert).

I guess we probably don't want to exclude Bob from this list in case Alice forgot that she already blocked Bob, and she's doesn't know why Bob isn't in the list of people to block...

One (stupid-simple-easy) solution would be to leave everyone in the list as is, but dismiss the view controller after selection. Rather than showing the "success" alert, the user could **see** that it worked via their updated blocked list. This would work OK as long as the user didn't have a ton of people in their blocked list, and could quickly see that Bob is in the blocked list.

Admittedly it would also be slower to block a lot of people at once with the proposed change, but I *think* that's an acceptable tradeoff for the greater good.

Also consider:

* adding an accessory view to indicate who is already blocked (or maybe a subtitle would be better yet).
* making the already-blocked contact rows unselectable.